apps/accounts/views.py

```diff
@@ -89,7 +92,12 @@ def inner(self, request):
                     user = request.user
             else:
                 user = identity_user
-            return fn(self, request, user=user, identity=identity)
+            if len(state_parts) == 2:
+                next_path = base64.urlsafe_b64decode(str(state_parts[1]))
```

Comment:
I think we had problems with unicode slugs in URLs in the past. I don't know if it's completely fixed so you might get a DecodeError here. You could try smart_decode() but that would probably just send them to a 404. I suggest:

1. Add some logging above this like 'about to redirect fxa user to path {}'
2. We just wait to see if we get any DecodeErrors. It's unlikely.